### PR TITLE
Added a Prefect example

### PR DESCRIPTION
**Pull request by zilto**
[Short description explaining the high-level reason for the pull request]

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

**Comment by zilto** [comment]
It seems that the pre-commit check is failing on files that were updated in the LLM example (independent from my file changes). What git command should I run to get the latest version of those on my branch?

File: examples/prefect/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+- [**Prefect**](https://prefect.io) is an open source orchestrator written in Python. Its purpose is to launch and execute your workflows over your infrastructure. It has integrations with many popular tools (Snowflake, dbt, Ray, OpenAI, etc.)
+- [**Hamilton**](https://github.com/dagworks-inc/hamilton) is a micro-framework to describe dataflows in Python. Its strength is expressing the flow of data & computation in a way that is straightforward to create, maintain, and resue (much like dbt for SQL)
+
+## Why both?
+Prefect includes useful features for production system such as job queues, caching, retries, logging, and more. However, each additional task you create comes with execution overhead and needs to be manually added to the DAG structure. Opting for less granular tasks ultimately reduces visibility and maintainability.
+
+In contrast, Hamilton automatically builds its DAG from the function definition, encouraging clean, small and single-purposed functions. By using Hamilton in a Prefect task, you gain this greater data lineage, reduce execution overhead, and get portable Python code you can run outside Prefect too.
```
Comment (skrawcz) [inline examples/prefect/README.md:10]:
```suggestion
In contrast, Hamilton automatically builds its DAG from the function definition, encouraging clean, small and single-purposed functions. By using Hamilton in a Prefect task, you gain this greater insight into lineage, reduced prefect execution overhead, and you get portable Python code you can run outside Prefect too.
```

**Review by skrawcz — COMMENTED** [review]
Looks good. Can you add information about what the code is doing/what it is for?  and mention that two tasks here are slightly arbitrary, they could be just one task, or more -- and Hamilton gives you the flexibility to not change too much code to do so.

File: examples/prefect/README.md (hunk 1 above)
Comment (zilto) [reply examples/prefect/README.md]: Fixed in the latest commits

File: examples/prefect/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
+# Hamilton + Prefect
+
+In this example, were going to show how to run a simple `data preprocessing -> model training -> model evaluation` workflow using Hamilton within Prefect tasks.
+- [**Prefect**](https://prefect.io) is an open source orchestrator written in Python. Its purpose is to launch and execute your workflows over your infrastructure. It has integrations with many popular tools (Snowflake, dbt, Ray, OpenAI, etc.)
```
Comment (skrawcz) [inline examples/prefect/README.md:4]:
```suggestion
- [**Prefect**](https://prefect.io) is an open source orchestrator written in Python. Its purpose is to launch and execute your workflows over your infrastructure. 
```

File: examples/prefect/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+- [**Hamilton**](https://github.com/dagworks-inc/hamilton) is a micro-framework to describe dataflows in Python. Its strength is expressing the flow of data & computation in a way that is straightforward to create, maintain, and reuse (much like dbt for SQL)
```
Comment (skrawcz) [inline examples/prefect/README.md:5]:
```suggestion
- [**Hamilton**](https://github.com/dagworks-inc/hamilton) is a micro-framework to describe dataflows in Python. Its strength is expressing the flow of data & computation in a way that is straightforward to create, maintain, and reuse (much like DBT has done for SQL). It is lightweight and can be run anywhere that python runs.
```

File: examples/prefect/README.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+## Why both?
+Prefect includes useful features for production system such as job queues, caching, retries, logging, and more. However, each additional task you create comes with execution overhead and needs to be manually added to the DAG structure. Opting for less granular tasks ultimately reduces visibility and maintainability.
```
Comment (skrawcz) [inline examples/prefect/README.md:8]:
```suggestion
Prefect is a heavy weight system that includes useful features for production system such as job queues, caching, retries, logging, and more. However, each additional "task" you create comes with execution overhead and needs to be manually added to the DAG structure. Opting for less granular tasks ultimately reduces visibility and maintainability, while opting for more creates more overhead.
```

File: examples/prefect/README.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+In contrast, Hamilton automatically builds its DAG from the function definition, encouraging clean, small and single-purposed functions. By using Hamilton in a Prefect task, you gain this greater insight into lineage, reduced Prefect execution overhead, and you get portable Python code you can run outside Prefect too.
```
Comment (skrawcz) [inline examples/prefect/README.md:10]:
```suggestion
In contrast, Hamilton automatically builds its DAG from the function definition itself, encouraging clean, small and single-purposed functions. By using Hamilton in a Prefect task, you gain this greater insight into lineage, reduced Prefect execution overhead and a simpler DAG to manage, and you get portable Python code you can run outside Prefect too.
```

File: examples/prefect/requirements.txt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+numpy
+pandas
+prefect
+scikit-learn
+sf-hamilton
+sf-hamilton[visualization]
```
Comment (skrawcz) [inline examples/prefect/requirements.txt:6]: Don't need both.
```suggestion
sf-hamilton[visualization]
```

File: examples/prefect/README.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+You'll notice that the Prefect workflow has 2 tasks: `prepare_data_task` and `train_and_evaluate_model_task` even though there are 3 function modules. While each function module best describes a logical grouping of functions, the Prefect tasks can help define how/where they should be executed. For example, the data processing task might require CPU resources while the model training and evaluation would benefit from having access to a GPU. This split allows us to benefit from Prefect's monitoring and tracking features, which make it easier to debug errors than having a single large Prefect task.
+
+For another workflow, a large Hamilton DAG in a single Prefect task, or smaller Prefect tasks and Hamilton modules might have been preferred. This shows how the Hamilton `Driver` brings flexibility to where and how you run your code.
+
+## Prefect setup
+The easiest way to get this example running is to sign up for Prefect's free tier and follow the [Prefect Cloud Quickstart](https://docs.prefect.io/latest/cloud/cloud-quickstart/) section. The steps are:
```
Comment (skrawcz) [inline examples/prefect/README.md:27]:
```suggestion
The easiest way to get this example running is to sign up for Prefect's free tier and follow the [Prefect Cloud Quickstart](https://docs.prefect.io/latest/cloud/cloud-quickstart/) section. Don't want to sign up? Don't worry, as Prefect is open source, can instead opt to host and run everything yourself as described [here](https://docs.prefect.io/2.11.0/getting-started/installation/). The steps are:
```

**Review by skrawcz — APPROVED** [review]

**Pull request merged**